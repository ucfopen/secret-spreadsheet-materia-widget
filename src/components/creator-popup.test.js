import React from 'react'
import Popup from './creator-popup'
import renderer from 'react-test-renderer'
import { shallow, mount } from '../enzyme'

describe('CreatorPopup component', function() {

	beforeEach(() => {
		jest.resetModules()
	})

	test('CreatorPopup renders popup with intro', () => {
		const props = {
			title: 'mock-title',
			showIntro: true,
			onSubmit: jest.fn(),
			onChange: jest.fn(),
		}

		const component = renderer.create(<Popup {... props}/>)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()

	})

	test('CreatorPopup renders popup', () => {
		const props = {
			title: 'tock-mitle',
			showIntro: false,
			onSubmit: jest.fn(),
			onChange: jest.fn(),
		}

		const component = renderer.create(<Popup {... props}/>)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()

	})

	test('CreatorPopup tests title text onChange', () => {
		const props = {
			title: 'lock-vital',
			showIntro: true,
			onSubmit: jest.fn(),
			onChange: jest.fn(),
		}

		const component = shallow(<Popup {...props}/>)
		component.find({type: 'text'}).simulate('change', {target: {value: 'dock-cycle'}})
		expect(props.onChange).toBeCalled()

	})

	test('CreatorPopup tests title form onSubmit', () => {
		const props = {
			title: 'sock-idol',
			showIntro: true,
			onSubmit: jest.fn(),
			onChange: jest.fn(),
		}

		const component = mount(<Popup {...props}/>)
		component.find({type: 'submit'}).simulate('submit')
		expect(props.onSubmit).toBeCalled()

	})

})
