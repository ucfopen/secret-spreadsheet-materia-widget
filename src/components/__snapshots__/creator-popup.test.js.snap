// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorPopup component CreatorPopup renders popup 1`] = `
<div
  className="popup"
>
  <div
    className="popup-inner"
  >
    <h2
      className="popup-hidden"
    >
      Secret Spreadsheet - How it works:
    </h2>
    <div
      className="popup-hidden"
    >
      <img
        src="assets/img/creator-tutorial.jpg"
      />
      <p>
        You'll create a spreadsheet and choose which cells to hide.
      </p>
      <p>
        Students will attempt to fill in the hidden cells correctly.
      </p>
    </div>
    <form
      className="title-container"
      onSubmit={[MockFunction]}
    >
      <label
        htmlFor="name-popup"
      >
        <strong>
          Give your Spreadsheet widget a title
        </strong>
      </label>
      <input
        autoFocus={true}
        className="popup-input"
        id="name-popup"
        onChange={[MockFunction]}
        placeholder="My Spreadsheet Widget"
        required="required"
        type="text"
        value="tock-mitle"
      />
      <input
        className="popup-submit"
        type="submit"
        value="Get Started"
      />
    </form>
  </div>
</div>
`;

exports[`CreatorPopup component CreatorPopup renders popup with intro 1`] = `
<div
  className="popup"
>
  <div
    className="popup-inner"
  >
    <h2
      className=""
    >
      Secret Spreadsheet - How it works:
    </h2>
    <div
      className=""
    >
      <img
        src="assets/img/creator-tutorial.jpg"
      />
      <p>
        You'll create a spreadsheet and choose which cells to hide.
      </p>
      <p>
        Students will attempt to fill in the hidden cells correctly.
      </p>
    </div>
    <form
      className="title-container"
      onSubmit={[MockFunction]}
    >
      <label
        htmlFor="name-popup"
      >
        <strong>
          Start by giving your Spreadsheet widget a title
        </strong>
      </label>
      <input
        autoFocus={true}
        className="popup-input"
        id="name-popup"
        onChange={[MockFunction]}
        placeholder="My Spreadsheet Widget"
        required="required"
        type="text"
        value="mock-title"
      />
      <input
        className="popup-submit"
        type="submit"
        value="Get Started"
      />
    </form>
  </div>
</div>
`;
